tests/tape/interfaces/test_qnode_jax.py

```diff
@@ -0,0 +1,205 @@
+
+    res = circuit(a, p)
+
+    expected = np.cos(a) * np.cos(p[1]) * np.sin(p[0]) + np.sin(a) * (
+        np.cos(p[2]) * np.sin(p[1]) + np.cos(p[0]) * np.cos(p[1]) * np.sin(p[2])
+    )
+    tol = 1e-5
+    assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    res = jax.grad(circuit, argnums=1)(a, p)
+    expected = np.array(
+        [
+            np.cos(p[1]) * (np.cos(a) * np.cos(p[0]) - np.sin(a) * np.sin(p[0]) * np.sin(p[2])),
+            np.cos(p[1]) * np.cos(p[2]) * np.sin(a)
+            - np.sin(p[1])
+            * (np.cos(a) * np.sin(p[0]) + np.cos(p[0]) * np.sin(a) * np.sin(p[2])),
+            np.sin(a)
+            * (np.cos(p[0]) * np.cos(p[1]) * np.cos(p[2]) - np.sin(p[1]) * np.sin(p[2])),
+        ]
+    )
+    assert np.allclose(res, expected, atol=tol, rtol=0)
+
+def qtransform(qnode, a, framework=jnp):
+    """Transforms every RY(y) gate in a circuit to RX(-a*cos(y))"""
+
+    def construct(self, args, kwargs):
+        """New quantum tape construct method, that performs
+        the transform on the tape in a define-by-run manner"""
+
+        # the following global variable is defined simply for testing
+        # purposes, so that we can easily extract the transformed operations
+        # for verification.
+
+        t_op = []
+
+        QNode.construct(self, args, kwargs)
+
+        new_ops = []
+        for o in self.qtape.operations:
+            # here, we loop through all tape operations, and make
+            # the transformation if a RY gate is encountered.
+            if isinstance(o, qml.RY):
+                t_op.append(qml.RX(-a * framework.cos(o.data[0]), wires=o.wires))
+                new_ops.append(t_op[-1])
+            else:
+                new_ops.append(o)
+
+        self.qtape._ops = new_ops
+        self.qtape._update()
+
+    import copy
+
+    new_qnode = copy.deepcopy(qnode)
+    new_qnode.construct = construct.__get__(new_qnode, QNode)
+    return new_qnode
+
+
+@pytest.mark.parametrize(
+    "dev_name,diff_method",
+    [("default.mixed", "finite-diff"), ("default.qubit.autograd", "parameter-shift")],
+)
+def test_transform(dev_name, diff_method, monkeypatch, tol):
+    """Test an example transform"""
+    monkeypatch.setattr(qml.operation.Operation, "do_check_domain", False)
+
+    dev = qml.device(dev_name, wires=1)
+
+    @qnode(dev, interface="jax", diff_method=diff_method)
+    def circuit(weights):
+        # the following global variables are defined simply for testing
+        # purposes, so that we can easily extract the operations for verification.
+        op1 = qml.RY(weights[0], wires=0)
+        op2 = qml.RX(weights[1], wires=0)
+        return qml.expval(qml.PauliZ(wires=0))
+
+    weights = np.array([0.32, 0.543])
+    a = np.array(0.5)
+
+    def loss(weights, a):
+        # the following global variable is defined simply for testing
+        # purposes, so that we can easily extract the transformed QNode
+        # for verification.

```

Comment:
Wait, which global variable?

Reply:
Oh, `new_circuit`, right?

Reply:
Yeah this comment should be deleted. new_circuit was treated as a global variable before and checked outside this method. However, we don't want to do that (the similar checks for the autograd interface don't apply here). I'll delete the comment.